lib/Parse/Lexer.cpp

```diff
@@ -1558,14 +1668,33 @@ void Lexer::tryLexEditorPlaceholder() {
 }
 
 StringRef Lexer::getEncodedStringSegment(StringRef Bytes,
-                                         SmallVectorImpl<char> &TempString) {
+                                         SmallVectorImpl<char> &TempString,
+                                         unsigned Modifiers,
+                                         const std::string &ToStrip) {
+
   TempString.clear();
   // Note that it is always safe to read one over the end of "Bytes" because
   // we know that there is a terminating " character.  Use BytesPtr to avoid a
   // range check subscripting on the StringRef.
   const char *BytesPtr = Bytes.begin();
-  while (BytesPtr != Bytes.end()) {
+  while (BytesPtr < Bytes.end()) {
     char CurChar = *BytesPtr++;
+
+    // Multiline string line ending normalization and indent stripping.
+    if (CurChar == '\r' || CurChar == '\n') {
+      bool stripNewline = (Modifiers & StringLiteralFirstSegment) &&
+        BytesPtr - 1 == Bytes.begin();
+      if (CurChar == '\r' && *BytesPtr == '\n')
+        BytesPtr++;
+      if (ToStrip.length() && StringRef(BytesPtr, ToStrip.length()) == ToStrip)
```

Comment:
Here, we only need to handle empty lines, right?  Any non-empty line will either have the right indent to strip, or we'd have already emitted an error about an invalid indent, right?